src/manifold/src/boolean3.cpp

```diff
@@ -137,7 +139,7 @@ inline thrust::pair<int, glm::vec2> Shadow01(
 
 // https://github.com/scandum/binary_search/blob/master/README.md
 // much faster than standard binary search on large arrays
-size_t monobound_quaternary_search(VecView<const int64_t> array, int64_t key) {
+ssize_t monobound_quaternary_search(VecView<const int64_t> array, int64_t key) {
```

Comment:
what is `ssize_t`? Okay, I found [this](https://stackoverflow.com/questions/19224655/using-ssize-t-vs-int) and am slightly more confused.

Reply:
`ssize_t` is signed version of `size_t`. But thinking about it, we only need that -1 to signal failure, and I can do it with `size_t` max as well.